src/Sentry.Unity/SentryMonoBehaviour.cs

```diff
@@ -120,7 +120,15 @@ internal ISentrySystemInfo SentrySystemInfo
 
     // Note: Awake is called only once and synchronously while the object is built.
     // We want to do it this way instead of a StartCoroutine() so that we have the context info ASAP.
-    private void Awake() => CollectData();
+    private void Awake()
+    {
+        CollectData();
+
+        // This prevents object from being destroyed when unloading the scene since using HideFlags.HideAndDontSave
+        // doesn't guarantee its persistence on all platforms i.e. WebGL
+        // (see https://github.com/getsentry/sentry-unity/issues/1678 for more details)
+        DontDestroyOnLoad(gameObject);
```

Comment:
```suggestion
        // This prevents object from being destroyed when unloading the scene since using HideFlags.HideAndDontSave
        // doesn't guarantee its persistence on all platforms i.e. WebGL
        // (see https://github.com/getsentry/sentry-unity/issues/1678 for more details)
        DontDestroyOnLoad(gameObject);
        
        CollectData();
```

Reply:
I think we can call this first. But that's just a random nitpick.